Source/RxPermission.swift

```diff
@@ -1,52 +1,85 @@
 //
-//  ViewController.swift
-//  RxPermission
-//
-//  Created by Sunshinejr on 04/16/2016.
-//  Copyright (c) 2016 Sunshinejr. All rights reserved.
+// RxPermission.swift
 //
+// Copyright (c) 2016 Sunshinejr <thesunshinejr@gmail.com>
+// 
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+// 
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
 
 import Permission
 import RxSwift
 
 public extension Permission {
-    
-    private struct AssociatedKeys {
-        static var Permission = "rx_permissionInstance"
-    }
-    
-    private func doLocked(closure: () -> Void) {
-        objc_sync_enter(self); defer { objc_sync_exit(self) }
-        closure()
-    }
-
-    var rx_permission: Observable<PermissionStatus> {
+    /// Reactive wrapper for `Permission` instance.
+    public var rx_permission: Observable<PermissionStatus> {
         return rx_permissionInstance.asObservable()
     }
-    
-    private var rx_permissionInstance: PublishSubject<PermissionStatus> {
+}
+
+// MARK: - Permission
+
+private extension Permission: AssociatedObject {
```

Comment:
We can't use private with protocol conformances 😢 I'd say we just move private to functions below?